File: 01-CORE-LEARNING/code/python/Python/template.py
#!/usr/bin/env python3
"""
Competitive Programming Template for Python
Author: Your Name
"""

import sys
import math
import bisect
from collections import defaultdict, deque, Counter
from heapq import heappush, heappop
from itertools import permutations, combinations, product
from functools import lru_cache

# Fast input
def fast_input():
    return sys.stdin.readline().strip()

def fast_int():
    return int(fast_input())

def fast_ints():
    return list(map(int, fast_input().split()))

def fast_float():
    return float(fast_input())

def fast_floats():
    return list(map(float, fast_input().split()))

# Constants
MOD = 10**9 + 7
INF = float('inf')
EPS = 1e-9

# Utility functions
def gcd(a, b):
    while b:
        a, b = b, a % b
    return a

def lcm(a, b):
    return a * b // gcd(a, b)

def power(base, exp, mod=MOD):
    """Fast exponentiation with modulo"""
    result = 1
    base = base % mod
    while exp > 0:
        if exp % 2 == 1:
            result = (result * base) % mod
        exp = exp >> 1
        base = (base * base) % mod
    return result

def is_prime(n):
    """Check if number is prime"""
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    for i in range(3, int(math.sqrt(n)) + 1, 2):
        if n % i == 0:
            return False
    return True

def sieve(n):
    """Sieve of Eratosthenes"""
    is_prime = [True] * (n + 1)
    is_prime[0] = is_prime[1] = False
    for i in range(2, int(math.sqrt(n)) + 1):
        if is_prime[i]:
            for j in range(i * i, n + 1, i):
                is_prime[j] = False
    return [i for i in range(n + 1) if is_prime[i]]

def binary_search(arr, target):
    """Binary search implementation"""
    left, right = 0, len(arr) - 1
    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    return -1

# Graph algorithms
def dfs(graph, start, visited=None):
    """Depth-First Search"""
    if visited is None:
        visited = set()
    visited.add(start)
    result = [start]
    for neighbor in graph[start]:
        if neighbor not in visited:
            result.extend(dfs(graph, neighbor, visited))
    return result

def bfs(graph, start):
    """Breadth-First Search"""
    visited = set()
    queue = deque([start])
    result = []
    
    while queue:
        vertex = queue.popleft()
        if vertex not in visited:
            visited.add(vertex)
            result.append(vertex)
            queue.extend(graph[vertex] - visited)
    return result

# Main function
def solve():
    """Main solving function"""
    # Read input
    n = fast_int()
    
    # Example: Read array
    arr = fast_ints()
    
    # Your algorithm implementation here
    
    # Output result
    print("Result:")

def main():
    """Main function for multiple test cases"""
    # For single test case, call solve() directly
    solve()
    
    # For multiple test cases:
    # t = fast_int()
    # for _ in range(t):
    #     solve()

if __name__ == "__main__":
    main()

File: 01-CORE-LEARNING/code/kotlin/Kotlin/template.kt
import java.io.*
import java.util.*
import kotlin.math.*

// Fast I/O
class FastScanner(stream: InputStream) {
    private val br = BufferedReader(InputStreamReader(stream))
    private var st: StringTokenizer? = null
    
    fun next(): String {
        while (st == null || !st!!.hasMoreTokens()) {
            st = StringTokenizer(br.readLine())
        }
        return st!!.nextToken()
    }
    
    fun nextInt() = next().toInt()
    fun nextLong() = next().toLong()
    fun nextDouble() = next().toDouble()
    fun nextLine() = br.readLine()!!
    
    fun nextInts(n: Int) = IntArray(n) { nextInt() }
    fun nextLongs(n: Int) = LongArray(n) { nextLong() }
    fun nextDoubles(n: Int) = DoubleArray(n) { nextDouble() }
}

class FastPrinter(stream: OutputStream) {
    private val pw = PrintWriter(stream)
    
    fun print(x: Any) = pw.print(x)
    fun println(x: Any) = pw.println(x)
    fun println() = pw.println()
    fun close() = pw.close()
    fun flush() = pw.flush()
}

// Constants
const val MOD = 1000000007L
const val INF = Long.MAX_VALUE
const val EPS = 1e-9

// Utility functions
fun gcd(a: Long, b: Long): Long = if (b == 0L) a else gcd(b, a % b)
fun lcm(a: Long, b: Long): Long = a / gcd(a, b) * b

fun power(base: Long, exp: Long, mod: Long = MOD): Long {
    var result = 1L
    var b = base % mod
    var e = exp
    while (e > 0) {
        if (e % 2 == 1L) {
            result = (result * b) % mod
        }
        e = e shr 1
        b = (b * b) % mod
    }
    return result
}

fun isPrime(n: Int): Boolean {
    if (n <= 1) return false
    if (n <= 3) return true
    if (n % 2 == 0 || n % 3 == 0) return false
    
    var i = 5
    while (i * i <= n) {
        if (n % i == 0 || n % (i + 2) == 0) {
            return false
        }
        i += 6
    }
    return true
}

fun sieve(n: Int): List<Int> {
    val isPrime = BooleanArray(n + 1) { true }
    isPrime[0] = false
    isPrime[1] = false
    
    for (i in 2..sqrt(n.toDouble()).toInt()) {
        if (isPrime[i]) {
            for (j in i * i..n step i) {
                isPrime[j] = false
            }
        }
    }
    
    return (2..n).filter { isPrime[it] }
}

fun binarySearch(arr: IntArray, target: Int): Int {
    var left = 0
    var right = arr.size - 1
    while (left <= right) {
        val mid = left + (right - left) / 2
        when {
            arr[mid] == target -> return mid
            arr[mid] < target -> left = mid + 1
            else -> right = mid - 1
        }
    }
    return -1
}

// Graph algorithms
fun dfs(graph: Array<MutableList<Int>>, start: Int, visited: BooleanArray, result: MutableList<Int>) {
    visited[start] = true
    result.add(start)
    
    for (neighbor in graph[start]) {
        if (!visited[neighbor]) {
            dfs(graph, neighbor, visited, result)
        }
    }
}

fun bfs(graph: Array<MutableList<Int>>, start: Int): List<Int> {
    val visited = BooleanArray(graph.size)
    val queue = ArrayDeque<Int>()
    val result = mutableListOf<Int>()
    
    queue.offer(start)
    visited[start] = true
    
    while (queue.isNotEmpty()) {
        val node = queue.poll()
        result.add(node)
        
        for (neighbor in graph[node]) {
            if (!visited[neighbor]) {
                visited[neighbor] = true
                queue.offer(neighbor)
            }
        }
    }
    return result
}

// Extension functions
fun IntArray.sum() = this.fold(0) { acc, x -> acc + x }
fun LongArray.sum() = this.fold(0L) { acc, x -> acc + x }
fun IntArray.product() = this.fold(1) { acc, x -> acc * x }
fun LongArray.product() = this.fold(1L) { acc, x -> acc * x }

fun <T> Array<T>.swap(i: Int, j: Int) {
    val temp = this[i]
    this[i] = this[j]
    this[j] = temp
}

fun IntArray.swap(i: Int, j: Int) {
    val temp = this[i]
    this[i] = this[j]
    this[j] = temp
}

// Solve function
fun solve(sc: FastScanner, out: FastPrinter) {
    // Read input
    val n = sc.nextInt()
    
    // Example: Read array
    val arr = sc.nextInts(n)
    
    // Your algorithm implementation here
    
    // Output result
    out.println("Result:")
}

fun main() {
    val sc = FastScanner(System.`in`)
    val out = FastPrinter(System.out)
    
    // For single test case
    solve(sc, out)
    
    // For multiple test cases:
    // val t = sc.nextInt()
    // repeat(t) {
    //     solve(sc, out)
    // }
    
    out.close()
}
